sourced/ml/transformers/basic.py

```diff
@@ -125,21 +125,33 @@ def __call__(self, head: RDD):
         return head.countApproxDistinct()
 
 
-class UastExtractor(Transformer):
-    def __init__(self, languages: Union[list, tuple], **kwargs):
+class LanguageSelector(Transformer):
+    def __init__(self, languages: Union[list, tuple], blacklist=False, **kwargs):
         super().__init__(**kwargs)
         self.languages = languages
+        self.blacklist = blacklist
 
     def __call__(self, files: DataFrame) -> DataFrame:
         files = files.dropDuplicates(("blob_id",)).filter("is_binary = 'false'")
         classified = files.classify_languages()
-        lang_filter = classified.lang == self.languages[0]
-        for lang in self.languages[1:]:
-            lang_filter |= classified.lang == lang
+        if not self.blacklist:
+            lang_filter = classified.lang == self.languages[0]
+            for lang in self.languages[1:]:
+                lang_filter |= classified.lang == lang
+        else:
+            lang_filter = classified.lang != self.languages[0]
+            for lang in self.languages[1:]:
+                lang_filter &= classified.lang != lang
         filtered_by_lang = classified.filter(lang_filter)
-        from pyspark.sql import functions
-        uasts = filtered_by_lang.extract_uasts().where(functions.size(functions.col("uast")) > 0)
-        return uasts
+        return filtered_by_lang
+
+
+class UastExtractor(Transformer):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __call__(self, files: DataFrame) -> DataFrame:
+        return files.extract_uasts().where(functions.size(functions.col("uast")) > 0)
```

Comment:
good to add an explanation why we need only uasts longer than 0. (if uast was extracted -> not empty list of results)